integration/cdi/src/main/java/org/apache/cxf/cdi/CdiServerConfigurableFactory.java

```diff
@@ -0,0 +1,85 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.cxf.cdi;
+
+import javax.enterprise.context.spi.CreationalContext;
+import javax.enterprise.inject.spi.AnnotatedType;
+import javax.enterprise.inject.spi.Bean;
+import javax.enterprise.inject.spi.BeanAttributes;
+import javax.enterprise.inject.spi.BeanManager;
+import javax.enterprise.inject.spi.InjectionTargetFactory;
+import javax.ws.rs.RuntimeType;
+import javax.ws.rs.core.Configurable;
+import javax.ws.rs.core.FeatureContext;
+
+import org.apache.cxf.jaxrs.impl.ConfigurableImpl;
+import org.apache.cxf.jaxrs.impl.ConfigurableImpl.Instantiator;
+import org.apache.cxf.jaxrs.provider.ServerConfigurableFactory;
+
+/** 
+ * Creates the instance of Configurable<?> suitable for CDI-managed runtime.
+ */
+public class CdiServerConfigurableFactory implements ServerConfigurableFactory {
+    private final BeanManager beanManager;
+    
+    CdiServerConfigurableFactory(final BeanManager beanManager) {
+        this.beanManager = beanManager;
+    }
+    
+    @Override
+    public Configurable<FeatureContext> create(FeatureContext context) {
+        return new CdiServerFeatureContextConfigurable(context, beanManager);
+    }
+    
+    /** 
+     * Instantiates the instance of the provider using CDI/BeanManager 
+     */
+    private static class CdiInstantiator implements Instantiator {
+        private final BeanManager beanManager;
+        
+        CdiInstantiator(final BeanManager beanManager) {
+            this.beanManager = beanManager;
+        }
+        
+        @Override
+        public <T> Object create(Class<T> cls) {
```

Comment:
I see we have a create... should we also have a destroy?  Otherwise, in CDI, you run the risk of having dependent scoped beans lying around.  I'm not sure if these are meant to be per-request objects or global objects, but if they are per request and `@Dependent` scoped then you'll want to destroy them.

Reply:
Was puzzled to have this as well but there is no lifecycle hook to call the destroy /dispose :(

Reply:
the client has close() so we can - worse case - use that. Also if the beanManager.isNormalScope(bean.getScope()) is true we don't need to call context.release()

Reply:
This is server-side ... :)

Reply:
-> bam :)

the server has it too (destroy() from memory)

Reply:
@rmannibucau  let me try to figure out if we could hook in there, thanks for the hint

Reply:
@johnament @rmannibucau Haven't found the acceptable way to dispose the created beans. It would be good to have that however `Feature`s (and `DynamicFeature`s) are instantiated at deployment (when JAX-RS resources are discovered) hereby should not be subject of request scoped but only global one. When server is shutting down (destroy) everything will be gone so may not worth introducing complex disposal mechanisms. Makes sense?

Reply:
Issue is more for dependent beans which must be released to avoid to leak data/memory when the app is undeployed but the server is not killed so we should really have a lifecycle for the configurable which is bound to the underlying instance and worse case to the Server.

Reply:
Hard to disagree with that indeed, CXF should be a good citizen there. So I've came up with quite simple solution using CDI events. Whenever we create the context in such circumstances, we register it for later disposal inside `JAXRSCdiResourceExtension` by firing the event (the same way we manage all other managed beans we create inside the CDI extension). It seems to work perfectly fine and releases the contexts on shutdown. Thoughts?

Reply:
Looks good for the server side :)

Reply:
Thanks guys for the comments, merging ...

Reply:
Hmmm I just noticed something looking at another issue.  I think this may assume that all providers now registered are within a CDI bean archive.  We may want to do a fallback in case the bean cannot be located.

Reply:
Thanks for spotting that @johnament, I will address it shortly.

Reply:
@johnament hmm (my turn), so in the test case we have there are 2 providers: one from bean archive, another from regular JAR file (not a bean archive). In both cases, the bean is located successfully. Looking why, BeanManager implementation (OWB, Weld) does the fallback in case there is no bean archive for a class available.

Reply:
@reta if not a bean you will not get a Bean<?> instance doing the lookup (empty Set). If it is found it means the jar was scanned and the bean activated somehow. The issue can come from the test setup, did you try with jackson for instance or a plain custom-provider-nocdi.jar?

Reply:
I did with `AtomFeedProvider` from plain no CDI archive, let me try Jackson to be sure :)

Reply:
@johnament No issues, works just fine for Jackson as well (no deployment archives are present but the bean was synthesized from the class).

Reply:
this is because you don't lookup a cdi bean but create one, this is probably not what we want. We should lookup an existing bean or fallback on creating a plain instance (take care of context injection support ;)). If we don't we can have side effects in applications duplicating beans during lookup or just pollute the CDI context for no reason.

Reply:
so basically the better way would be to call beanManager.getBeans() to ensure there is one in the context, and fallback to createBean if none, right? :-)

Reply:
nop, try getbeans and if resolve() returns null then use standalone logic (default newInstance())

Reply:
sounds good, thanks @rmannibucau